pkg/sbom/node.go

```diff
@@ -291,11 +291,36 @@ func (n *Node) Purl() PackageURL {
 		return ""
 	}
 
-	for _, e := range n.ExternalReferences {
+	for _, e := range n.Identifiers {
 		if e.Type == "purl" {
-			return PackageURL(e.Url)
+			return PackageURL(e.Value)
 		}
 	}
 
 	return ""
 }
+
+// HashesMatch takes a map of hashes and returns a boolean indicating
+// if the test hashes th match the nodes hashes. The algorithm will
```

Comment:
```suggestion
// if the test hashes to match the nodes hashes. The algorithm will
```

Reply:
Ignore my suggested commits here and below if `th` is a variable reference. But perhaps consider another way to reference this variable. I was confused while reading this comment.

Reply:
I reworded the whole comment because I could only read "puerco should sleep instead of writing this comment". I hope it is clearer now.